core/src/test/scala/org/apache/spark/sql/delta/DeltaSourceSuite.scala

```diff
@@ -481,6 +518,43 @@ class DeltaSourceSuite extends DeltaSourceSuiteBase with DeltaSQLCommandTest {
     }
   }
 
+  test("maxBytesPerTrigger: Trigger.AvailableNow respects read limits") {
+    withTempDir { inputDir =>
+      val deltaLog = DeltaLog.forTable(spark, new Path(inputDir.toURI))
+      (0 until 5).foreach { i =>
+        val v = Seq(i.toString).toDF
+        v.write.mode("append").format("delta").save(deltaLog.dataPath.toString)
+      }
+
+      val stream = spark.readStream
+        .format("delta")
+        .option(DeltaOptions.MAX_BYTES_PER_TRIGGER_OPTION, "1b")
+        .load(inputDir.getCanonicalPath)
+        .writeStream
+        .format("memory")
+        .trigger(Trigger.AvailableNow)
+        .queryName("maxBytesPerTriggerTest")
+
+      var q = stream.start()
+      try {
+        assert(q.awaitTermination(10000))
+        val progress = q.recentProgress.filter(_.numInputRows != 0)
+        assert(progress.length === 5)
+        progress.foreach { p =>
+          assert(p.numInputRows === 1)
+        }
+        checkAnswer(sql("SELECT * from maxBytesPerTriggerTest"), (0 until 5).map(_.toString).toDF)
+
+        // Restarting the stream should immediately terminate with no progress because no more data
+        q = stream.start()
+        assert(q.awaitTermination(2000))
```

Comment:
2000 maybe a little less ... can cause flakiness. maybe make it 10 second?

Reply:
Yeah I wasn't sure what to pick. 10 sounds good

Reply:
Updated to 10s